PxGraf/Models/Metadata/HeaderBuildingUtilities.cs

```diff
@@ -0,0 +1,185 @@
+                }
+            }
+            return header;
+        }
+
+        private static void AppendTimeValuePlaceholders(this StringBuilder builder, Translation translation, bool multivalue)
+        {
+            if (builder.Length > 0) builder.Append(' ');
+
+            if (multivalue)
+            {
+                builder.Append(string.Format(translation.HeaderTimeFormats.TimeRange, "[FIRST]", "[LAST]"));
+            }
+            else
+            {
+                builder.Append(string.Format(translation.HeaderTimeFormats.SingleTimeValue, "[FIRST]"));
+            }
+        }
+
+        private static void AppendContentDimensionText(this Dictionary<string, StringBuilder> builders, IReadOnlyDimension contentDim, string language, MatrixQuery? query)
+        {
+            if (contentDim.Values.Count == 1)
+            {
+                builders[language].Append(GetDimensionValueNameEditForLanguage(query, contentDim.Code, contentDim.Values[0], language));
+            }
+            else
+            {
+                builders[language].Append(GetDimensionNameEditForLanguage(query, contentDim, language));
+            }
+        }
+
+        private static void AppendSingleValueDimensionTexts(this Dictionary<string, StringBuilder> builders, IReadOnlyList<IReadOnlyDimension> dimensions, string language, MatrixQuery? query)
+        {
+            IEnumerable<IReadOnlyDimension> singleValueDimensions = dimensions
+                .Where(d => d.Values.Count == 1 && d.Type != DimensionType.Content && d.Type != DimensionType.Time);
+
+            foreach (var singleDim in singleValueDimensions)
+            {
+                string? eliminationCode = singleDim.GetEliminationValueCode();
+                IReadOnlyDimensionValue dimensionValue = singleDim.Values[0];
+                if (dimensionValue.Code != eliminationCode)
+                {
+                    if (builders[language].Length > 0)
+                    {
+                        builders[language].Append(", ");
+                    }
+
+                    builders[language].Append(GetDimensionValueNameEditForLanguage(query, singleDim.Code, dimensionValue, language));
+                }
+            }
+        }
+
+        private static string GetDimensionNameEditForLanguage(MatrixQuery? query, IReadOnlyDimension dimension, string language)
+        {
+            if (query?.DimensionQueries?.TryGetValue(dimension.Code, out DimensionQuery? dq) == true &&
+                dq.NameEdit?.Languages.Contains(language) == true)
+            {
+                return dq.NameEdit[language];
+            }
+            return dimension.Name[language];
+        }
+
+        private static string GetDimensionValueNameEditForLanguage(MatrixQuery? query, string dimensionCode, IReadOnlyDimensionValue value, string language)
+        {
+            if (query?.DimensionQueries?.TryGetValue(dimensionCode, out DimensionQuery? dq) == true &&
```

Comment:
== true is redundant

Reply:
So I thought as well, but in this specific case they are nullable booleans which can't be compared directly